src/morphometrics/segment/segment.py

```diff
@@ -0,0 +1,106 @@
+import glob
+import os
+
+import h5py
+import numpy as np
+import tifffile
+import wget
+import yaml
+from plantseg.pipeline.raw2seg import raw2seg
+from stardist.models import StarDist3D
+
+
+def load_to_disk(preprocessed_image):
+    """
+    If args.model == plant-seg,
+    create a directory and write the preprocessed image in it, for raw2seg() to call (raw2seg() requires to load images from disk).
+
+    Parameters
+    ----------
+    preprocessed_image: np.ndarray
+        the preprocessed image
+
+    Returns
+    -------
+    preprocessed_path: string
+        path to the preprocessed images on the disk ('./preprocessed')
+    """
+
+    parent_dir = os.getcwd()
+    directory = "preprocessed"
+    preprocessed_path = os.path.join(parent_dir, directory)
+    if not os.path.exists(preprocessed_path):
+        os.mkdir(preprocessed_path)
+
+    tifffile.imwrite(f"{preprocessed_path}/temp.tif", preprocessed_image)
+
+    return preprocessed_path
+
+
+def make_config(preprocessed_image):
+    """
+    If args.model == plant-seg,
+    make the plantseg default yaml or local (with file name "config.yaml") configuration.
+
+    Parameters
+    -----------
+    preprocessed_image: np.ndarray
+        the preprocessed image
+
+    Returns
+    -----------
+    config: Python object corresponding to the defined configuration
+        the user-defined configuration
+    """
+
+    file = r".\config.yaml"
+    if not os.path.exists(file):
+        # if no local configuration, load the plant-seg example yaml file from github
+        url = "https://raw.githubusercontent.com/hci-unihd/plant-seg/master/examples/config.yaml"
```

Comment:
For reproducibility, I think we should save the config and upload it to our repo. That way we don't get impacted if they change the config on their repo. maybe put it in a `morphometrics/segmentation/configs`?